pkg/storage/v2/mysql/query.go

```diff
@@ -360,6 +360,21 @@ func ConstructQueryAndWhereArgs(baseQuery string, options *ListOptions) (query s
 	return
 }
 
+// ConstructQueryAndWhereArgsOnlyUseWherePart builds a query with only the WHERE part.
+// This Function is intended to be used when the purpose is
+// to generate a query that does not require any conditions other than the Where clause,
+// such as "SELECT COUNT(1) FROM XXX".
+func ConstructQueryAndWhereArgsOnlyUseWherePart(
```

Comment:
The name ConstructQueryAndWhereArgsOnlyUseWherePart is too verbose and not very clear.
WDYT?

```suggestion
// ConstructCountQuery builds a count query with optional filtering.
// It takes a base count SQL query and ListOptions, and returns the complete query and its arguments.
// The base SQL should be a valid COUNT query (e.g., "SELECT COUNT(1) FROM table").
// Additional WHERE clauses from ListOptions will be appended to the base query.
func ConstructCountQuery(
```

Reply:
@cre8ivejp 
There was a concern that `ConstructCountQuery()` would give the impression that `COUNT(1)` was added to the query within the function,
so we gave it this name to distinguish it from the original `ConstructQueryAndWhereArgs().`
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/storage/v2/mysql/query.go#L348

This name is certainly long, so how about setting a flag to` ConstructQueryAndWhereArgs()` to branch the processing as shown below?

```
func ConstructQueryAndWhereArgs(baseQuery string, options *ListOptions, bool isOnlyUseWhereParts) (query string, whereArgs []interface{})
```

Reply:
@cre8ivejp 
If I think about it, this function constructs a query based on ListOption for the base query, so I think `ConstructsCountQuery` would be fine.
I'm going to fix that.

Reply:
I changed to `ConstructCountQuery`.
Please check again.